source/dem/plane_insertion.cc

```diff
@@ -0,0 +1,281 @@
+      // empty_cells_on_proc needs to be shorter for some processor.
+      while (empty_cells_on_proc.size() <
+             number_of_particles_to_insert_per_core[this_mpi_process])
+        {
+          auto it = empty_cells_on_proc.end();
+          empty_cells_on_proc.erase(it);
+        }
+
+      // There's probably a better ways to define the properties of a particle,
+      // but for now it works...
+      double type     = this->current_inserting_particle_type;
+      double diameter = dem_parameters.lagrangian_physical_properties
+                          .particle_average_diameter.at(type);
+      double density =
+        dem_parameters.lagrangian_physical_properties.density_particle.at(type);
+      double vel_x        = dem_parameters.insertion_info.vel_x;
+      double vel_y        = dem_parameters.insertion_info.vel_y;
+      double vel_z        = dem_parameters.insertion_info.vel_z;
+      double omega_x      = dem_parameters.insertion_info.omega_x;
+      double omega_y      = dem_parameters.insertion_info.omega_y;
+      double omega_z      = dem_parameters.insertion_info.omega_z;
+      double fem_force_x  = 0.;
+      double fem_force_y  = 0.;
+      double fem_force_z  = 0.;
+      double fem_torque_x = 0.;
+      double fem_torque_y = 0.;
+      double fem_torque_z = 0.;
+      double mass         = density * 4. / 3. * M_PI *
+                    Utilities::fixed_power<3, double>(diameter * 0.5);
+      double volumetric_contribution = 0.;
+
+      std::vector<double> properties_of_one_particle{type,
+                                                     diameter,
+                                                     vel_x,
+                                                     vel_y,
+                                                     vel_z,
+                                                     omega_x,
+                                                     omega_y,
+                                                     omega_z,
+                                                     fem_force_x,
+                                                     fem_force_y,
+                                                     fem_force_z,
+                                                     fem_torque_x,
+                                                     fem_torque_y,
+                                                     fem_torque_z,
+                                                     mass,
+                                                     volumetric_contribution};
+
+      // Loop over the empty cells we kept.
+      for (const auto &cell : empty_cells_on_proc)
+        {
+          // Position of the center of the cell
+          Point<dim> insertion_location =
+            cells_centers.at(cell->active_cell_index());
+
+          // Generate the random point of insertion
+          insertion_location(0) +=
+            static_cast<double>(rand()) / static_cast<double>(RAND_MAX) *
```

Comment:
Maybe you can have a variable =
            dem_parameters.insertion_info.random_number_range / static_cast<double>(RAND_MAX)
to multiply to each particle location random number since it's a constant